### PR TITLE
Add initial implementation for message-observer

### PR DESCRIPTION
**Pull request by ermyas**
This PR adds an initial implementation of a Relayer's [message-observer component](https://github.com/ConsenSys/gpact/blob/main/doc/relayer-design.md). This component is responsible for listening to configured events, transforming them into the common message format and then enqueueing the messages onto a queue for further processing by other Relayer components.

### CODE REVIEW

**Review by drinkcoffee — CHANGES_REQUESTED** [review]
I have put some comments in for changes.

File: messaging/relayer/internal/msgobserver/eth/soliditywrappers/sfc/simple_crosschain_control.go [hunk 1]
```diff
@@ -0,0 +1,899 @@
+// Code generated - DO NOT EDIT.
```
[the remark this answers is not in the document]
Comment (ermyas) [reply messaging/relayer/internal/msgobserver/eth/soliditywrappers/sfc/simple_crosschain_control.go:1]: good point

[the remark this answers is not in the document]
Comment (ermyas) [reply messaging/relayer/internal/msgobserver/eth/soliditywrappers/sfc/simple_crosschain_control.go]: moved the `soliditywrappers` folder outside of the message observer

File: messaging/relayer/internal/msgobserver/eth/utils.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package eth
+
+import (
+	"crypto/rand"
```
[the remark this answers is not in the document]
Comment (ermyas) [reply messaging/relayer/internal/msgobserver/eth/utils.go]: updated to use our prng

**Review by wcgcyx — APPROVED** [review]

**Review by drinkcoffee — APPROVED** [review]

**Pull request merged**